packages/react/menu/src/Menu.tsx

```diff
@@ -224,16 +228,36 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
 /* ---------------------------------------------------------------------------------------------- */
 
 type MenuRootContentOwnProps = Omit<
-  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  Polymorphic.OwnProps<typeof MenuRootContentModal | typeof MenuRootContentNonModal>,
   keyof MenuContentImplPrivateProps
 >;
 type MenuRootContentPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  Polymorphic.IntrinsicElement<typeof MenuRootContentModal | typeof MenuRootContentNonModal>,
   MenuRootContentOwnProps
 >;
 
 const MenuRootContent = React.forwardRef((props, forwardedRef) => {
   const context = useMenuContext(CONTENT_NAME);
+
+  return context.modal ? (
+    <MenuRootContentModal {...props} ref={forwardedRef} />
+  ) : (
+    <MenuRootContentNonModal {...props} ref={forwardedRef} />
```

Comment:
Same as prior implementations. I think this is still the cleanest way to provide the two modes though I am a little concerned at the size of `Menu`